### ISSUE TITLE
RequireThis reports false-positive when an instance method is overloaded with a static one

### ISSUE THREAD

**Issue by Vladlis** [checkstyle/checkstyle#2539]
Input:
```java
public class Test {
	
	public void foo() {}
	
	public static void foo(int i) {}
	
	public void bar() {
		foo();		// line 8
		foo(1);     // line 9
	}
}
```
Config:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    
    <module name="TreeWalker">
         <module name="RequireThisCheck"/>
    </module>
</module>
```
Result:
```
Starting audit...
D:\cli\Test.java:8:17: warning: Method call to 'foo' needs "this.".
D:\cli\Test.java:9:17: warning: Method call to 'foo' needs "this.".       <-- false-positive
Audit done.
```

The reason is searching for a called method just by name.

**Comment by Vladlis** [comment]
As it was discussed in https://github.com/checkstyle/checkstyle/pull/2588 adding "this." to method calls does not make much of a sense. So we should decide:
1) Whether to keep "checkMethods" option?
2) If yes, than should this bug be fixed?

**Comment by romani** [comment]
give some insight why some people use "this." for methods:
http://stackoverflow.com/questions/146989/is-writing-this-before-instance-variable-and-methods-good-or-bad-style

Summary:
1) The same notation/habit for C++ and Java (C++ have global methods, so having "this." do make sense in it to distinguish call of method of class instead of global).
2) non IDE development (ease of refactoring, some clearness to distinguish static and non static methods)

both of these reasons is not good for Java at current days. 
We already have this functionality and it is activated by default, we need to support it.

Believe it or not , but people do use this Check:
https://github.com/search?l=xml&q=RequireThis&type=Code&utf8=%E2%9C%93
https://github.com/search?l=xml&q=RequireThisCheck&type=Code&utf8=%E2%9C%93

Resolution: Issue need to be fixed.

**Comment by romani** [comment]
as we finally found a reason of this Check , please update its description to describe "Rationale: ..... "
It might be useful to put a warning at the top that such Check is very controversial and not that actual at current days.

**Comment by romani** [comment]
fix is merged.
